shotover-proxy/src/message/mod.rs

```diff
@@ -130,6 +130,14 @@ impl Message {
         }
     }
 
+    pub(crate) fn get_raw_bytes(&self) -> &Bytes {
+        match self.inner.as_ref().unwrap() {
+            MessageInner::RawBytes { bytes, .. } => bytes,
+            MessageInner::Parsed { bytes, .. } => bytes,
+            MessageInner::Modified { .. } => panic!("can't get raw bytes of modified"),
```

Comment:
I do not like the idea of a public method on Message that just panics if the message is in a certain state.

Reply:
Yes, I'm intending to change this. Apologies, I should've marked as a draft.

Reply:
Not sure what the best approach here is. I only want to return a reference to the bytes because this function is used in a very hot path so cloning the bytes so we can own it is not possible, but to return the `bytes` for a `MessageInner::Modified` means we have to encode it and then that value will be owned by the function.

Some options:
- Leaving the `panic` in here, keeping it as `pub (crate)` so it can't be used by transform developers and marking it clearly with a comment that this should not be used where it's possible the message might be in that state.
- Making `inner` `pub (crate)` so I can write this logic in the codec to ensure it's only used there.

Reply:
I'm confused your comment mentions using it in a codec, but the only place its used is in a transform.

Reply:
Oops, updated my comment.

Reply:
A possible option is making a reference version of  `fn into_encodable(self, expected_message_type: MessageType) -> Result<Encodable> {`

Something like: `as_encodable(self, expected_message_type) -> Result<EncodableRef>` 
where:
```rust
enum EncodableRef {
    Bytes(&Bytes),
    Frame(&Frame),
}
```

If we did that we should probably rename Encodable to something more generic though.